QWeb/internal/xhr.py

```
@@ -16,12 +16,256 @@
 # ---------------------------
 import time
 from robot.api import logger
-from selenium.common.exceptions import WebDriverException, JavascriptException
+from selenium.common.exceptions import JavascriptException, WebDriverException
+from typing import Optional
+from QWeb.internal import javascript, util
 from QWeb.internal.exceptions import QWebDriverError
-from QWeb.internal import javascript
+from QWeb.keywords import config
+
+
+# Install monitor (fetch/XMLHttpRequest + MutationObserver). Idempotent.
+JS_INSTALL_MONITOR = r"""
+return (function () {
+  if (window.__xhrMon && window.__xhrMon.installed) return true;
+
+  window.__xhrMon = Object.assign(window.__xhrMon || {}, {
+    installed: true,
+    pending: (window.__xhrMon && typeof window.__xhrMon.pending === "number") ? window.__xhrMon.pending : 0,
+    fetchPatched: false,
+    xhrPatched: false,
+    lastMutationTs: Date.now(),
+    observerStarted: false
+  });
+
+  try {
+    if (!window.__xhrMon.fetchPatched && typeof window.fetch === "function") {
+      const _orig = window.fetch;
+      window.fetch = function() {
+        try { window.__xhrMon.pending++; } catch(e) {}
```

Silent error discarding of all errors seems bit questionable.

I don't know how often and what errors occur here and if they clog the console but some handling could be beneficial.
I'm guessing that the `++` operation is quite likely to fail which throws and can be silently ignored, but other, more rare and yet unidentified, errors probably should get logged somewhere

---

One option would be passing a debug flag into the functions to enable collecting and returning errors only when needed.